lib/systems/default.nix

```
@@ -158,38 +158,47 @@ rec {
         if final.isMacOS then "MACOSX_DEPLOYMENT_TARGET"
         else if final.isiOS then "IPHONEOS_DEPLOYMENT_TARGET"
         else null;
+    } // (
+      let
+        emulators = pkgs:
+          let
+            qemu-user = pkgs.qemu.override {
+              smartcardSupport = false;
+              spiceSupport = false;
+              openGLSupport = false;
+              virglSupport = false;
+              vncSupport = false;
+              gtkSupport = false;
+              sdlSupport = false;
+              pulseSupport = false;
+              smbdSupport = false;
+              seccompSupport = false;
+              hostCpuTargets = [ "${final.qemuArch}-linux-user" ];
+            };
+            wine-name = "wine${toString final.parsed.cpu.bits}";
+            wine = (pkgs.winePackagesFor wine-name).minimal;
+          in
+          if final.parsed.kernel.name == pkgs.stdenv.hostPlatform.parsed.kernel.name &&
+            pkgs.stdenv.hostPlatform.canExecute final
+          then "${pkgs.runtimeShell} -c '\"$@\"' --"
+          else if final.isWindows
+          then "${wine}/bin/${wine-name}"
+          else if final.isLinux && pkgs.stdenv.hostPlatform.isLinux
+          then "${qemu-user}/bin/qemu-${final.qemuArch}"
+          else if final.isWasi
+          then "${pkgs.wasmtime}/bin/wasmtime"
+          else if final.isMmix
+          then "${pkgs.mmixware}/bin/mmix"
+          else null;
+      in {
+        emulator = pkgs:
+          if (final.emulatorAvailable pkgs)
+          then emulators pkgs
+          else throw "Don't know how to run ${final.config} executables.";
```

```suggestion
        emulatorAvailable = pkgs: !lib.isNull (emulators pkgs);

        emulator = pkgs:
          if (final.emulatorAvailable pkgs)
          then emulators pkgs
          else throw "Don't know how to run ${final.config} executables.";
```

(and delete line below, GH won't let me do that in a review suggestion.)

---

done but did `(selectEmulator pkgs) != null` instead because `isNull` is "deprecated" (for 10+ years...)